python/setup.py

```diff
@@ -100,13 +100,16 @@
     os.walk("ray/new_dashboard/client/build") for filename in filenames
 ]
 
+# Base dependencies for extras.
+base_deps = ["requests"]
+
 # If you're adding dependencies for ray extras, please
 # also update the matching section of requirements/requirements.txt
 # in this directory
 extras = {
-    "default": ["colorful"],
-    "serve": ["uvicorn", "requests", "starlette", "fastapi"],
-    "tune": ["pandas", "tabulate", "tensorboardX>=1.9"],
+    "default": base_deps + ["colorful"],
+    "serve": base_deps + ["uvicorn", "starlette", "fastapi"],
+    "tune": base_deps + ["pandas", "tabulate", "tensorboardX>=1.9"],
```

Comment:
This is a bit clumsy (introduces extra redirection) -- doesn't hurt to just include it inline imo!

Reply:
changed